pkg/path/walk.go

```diff
@@ -27,43 +28,49 @@ func GetPipelinePaths(root, pipelineDefinitionFile string) ([]string, error) {
 			return nil
 		}
 
-		if !strings.HasSuffix(path, pipelineDefinitionFile) {
-			return nil
-		}
+		// Check if the file matches any of the pipeline definition file names
+		for _, pipelineDefinition := range pipelineDefinitionFile {
+			if strings.HasSuffix(path, pipelineDefinition) {
+				abs, err := filepath.Abs(path)
+				if err != nil {
+					return errors.Wrapf(err, "failed to get absolute path for %s", path)
+				}
 
-		abs, err := filepath.Abs(path)
-		if err != nil {
-			return errors.Wrapf(err, "failed to get absolute path for %s", path)
+				pipelinePaths = append(pipelinePaths, filepath.Dir(abs))
+				break // Stop checking other file names if one matches
+			}
 		}
 
-		pipelinePaths = append(pipelinePaths, filepath.Dir(abs))
-
 		return nil
 	})
+
 	if err != nil {
 		return nil, errors.Wrapf(err, "error walking directory")
 	}
 
 	return pipelinePaths, nil
 }
 
-func GetPipelineRootFromTask(taskPath, pipelineDefinitionFile string) (string, error) {
+func GetPipelineRootFromTask(taskPath string, pipelineDefinitionFile []string) (string, error) {
 	absoluteTaskPath, err := filepath.Abs(taskPath)
 	if err != nil {
 		return "", errors.Wrapf(err, "failed to convert task path to absolute path")
 	}
 
 	currentFolder := absoluteTaskPath
-	for currentFolder != filepath.VolumeName(currentFolder)+string(os.PathSeparator) && currentFolder != "/" {
-		tryPath := filepath.Join(currentFolder, pipelineDefinitionFile)
-		if _, err := os.Stat(tryPath); err == nil {
-			return currentFolder, nil
+	rootPath := filepath.VolumeName(currentFolder) + string(os.PathSeparator)
+	for currentFolder != rootPath && currentFolder != "/" {
+		for _, pipelineDefinition := range pipelineDefinitionFile {
+			tryPath := filepath.Join(currentFolder, pipelineDefinition)
+			if _, err := os.Stat(tryPath); err == nil {
```

Comment:
I think `os.Stat(tryPath)` might not be the perfect way to check for existence. There could be a different error from "does not exist" 
See answer here https://stackoverflow.com/questions/12518876/how-to-check-if-a-file-exists-in-go

Reply:
Our existing implementation was using this line `os.Stat(tryPath)` that's why I kept it